homeassistant/components/homekit_controller/humidifier.py

```diff
@@ -0,0 +1,224 @@
+"""Support for HomeKit Controller humidifier."""
+from typing import Any, Dict, List, Optional
+
+from aiohomekit.model.characteristics import CharacteristicsTypes
+
+from homeassistant.components.humidifier import HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    ATTR_AVAILABLE_MODES,
+    ATTR_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MODE,
+    DEVICE_CLASS_DEHUMIDIFIER,
+    DEVICE_CLASS_HUMIDIFIER,
+    SUPPORT_MODES,
+)
+from homeassistant.core import callback
+
+from . import KNOWN_DEVICES, HomeKitEntity
+
+SUPPORT_FLAGS = 0
+
+ATTR_DEHUMIDIFIER_THRESHOLD = "dehumidifier_threshold"
+ATTR_HUMIDIFIER_THRESHOLD = "humidifier_threshold"
+ATTR_CURRENT_HUMIDITY = "current_humidity"
+
+HK_MODE_TO_HA = {
+    0: "off",
+    1: "auto",
+    2: "humidifying",
+    3: "dehumidifying",
+}
+
+HA_MODE_TO_HK = {
+    "auto": 0,
+    "humidifying": 1,
+    "dehumidifying": 2,
+}
```

Comment:
Something I don't get: HA modes should be from the available modes in this file, right ?
 https://github.com/home-assistant/core/blob/bdb71c9159621fb17a7cc66a825585d486b5c23d/homeassistant/components/humidifier/const.py#L3-L11

Reply:
I wasn't really sure what to do since the modes in HomeKit are not available in the the `humdifier/const.py`. Am I allowed to just add them to the file in the PR, or keep them near the top of this file.

Reply:
![IMG_A2A877C54A0E-1](https://user-images.githubusercontent.com/1464185/97097754-f6dcf980-164a-11eb-8ba4-ec9bf18f1618.jpeg)

Reply:
Ok, I did not see the duality of the combo device.

Hard to map indeed.